VIPs/vip-7.md

```diff
@@ -0,0 +1,294 @@
+
+- Latency for the current user
+- Average latency for all users
+- Upload and download speed
+
+### Recommendations
+
+1. Verida client implementation will provide a method to easily obtain the `latency` for a storage node (measured in milliseconds).
+2. Providing dynamic feedback on uptime and outages is a non-trivial process, so `average latency` and `up / down speed` is out of scope for now.
+
+## Regulatory jurisdiction
+
+The regulatory jurisdiction of a storage node could consider the following relevant metadata:
+
+- Country where the storage node is located
+- Country of legal entity operating the storage node
+
+Some governments have the right to access digital infrastructure if it is located in a particular country. Similarly some governments have the right to access digital infrastructure, regardless of country the data is located, if the infrastructure operator falls under their jurisdiction.
+
+### Recommendations
+
+1. Require storage nodes to specify their `countryResidence`. This information will not be used other than to display to the user.
+2. Require storage nodes to specify their `countryLocation` (as per geographic region above). This will be enabled by the storage node specifying their `datacenter`, which includes the `countryLocation`.
+
+## Node operator
+
+It may be preferable for Users to have nodes operated by different operators. ie: You may not want all your nodes operated by a particular organization.
+
+## User preference
+
+Users may have specific preferences. Here are some examples:
+
+1. I only care about having the fastest possible experience
+2. I don't trust my government, so want to ensure my data is stored in another country, regardless of how slow it is
+3. I want to maximise redundancy, so want to choose storage nodes across 5 different countries
+
+The protocol will focus on providing sensible default for all users, but it will be possible for advanced users to completely customize where their data is stored.
+
+# Specification
+
+## Discovery
+
+### Storage Nodes
+
+The list of verified storage node operators will be maintained in a JSON file at a known URL (`Storage Node Registry`). This URL endpoint will be updated by the Verida Foundation and embedded in the Verida protocol client implementation.
+
+This file will be located at:
+
+1. https://meta.verida.network/registry/storageNodes/testnet.json
+2. https://meta.verida.network/registry/storageNodes/mainnet.json
+
+Example file:
+
+```
+[
+    {
+        "id": "verida-testnet-aws-us-east1-001",
+        "name": "Verida Foundation Testnet: AWS-US-EAST1-001",
+        "description": "Node operated by the Verida Foundation on Testnet",
+        "datacenter": "aws-us-east1",
+        "serviceEndpoint": "https://001-use1.tn.verida.tech/",
+        "establishmentDate": "2023-01-03T08:22:35Z",
+        "countryResidence": "SG"
+    },
+    ...
+]
+```
+
+- `id`: Unique alphanumeric string that identifies the storage node
```

Comment:
Can it be a uuid rather than a meaningful id to avoid duplicate issues?

Reply:
Good point.